File: 2018-02-20-352-easy-making-imgur-style-links/main.cpp
#include <iostream>
#include <string>

int main(int argc, char *argv[])
{
	std::string line;
	const std::string digits = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
	while (std::getline(std::cin, line)) {
		unsigned long long n = std::stoull(line);
		std::string result = "";
		while (n > 0) {
			auto offset = n % static_cast<unsigned long long>(digits.length());
			std::cout << "Offset: " << offset << '\n';
			result = digits.at(offset) + result;
			n /= 62ULL;
		}
		std::cout << result << '\n';
	}
	return 0;
}

File: 2018-02-20-352-easy-making-imgur-style-links/main.swift
let input = [187621, 237860461, 2187521, 18752]

func printBase62(_ n: Int) {
    let digits = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
    var n_mut: Int = n
    var result = String()
    while (n_mut > 0) {
        let offset = n_mut % 62
        let index = digits.index(digits.startIndex, offsetBy: offset)
        result = String(digits[index]) + result
        n_mut /= 62
    }
    print(result)
}

input.forEach {
    printBase62($0)
}
